ksqldb-engine/src/main/java/io/confluent/ksql/engine/QueryCleanupService.java

```diff
@@ -91,20 +92,27 @@ public void addCleanupTask(final QueryCleanupTask task) {
 
   public static class QueryCleanupTask implements Runnable {
     private final String appId;
+    private final String queryTopicPrefix;
+    private final Optional<String> topologyName;
+    private final String pathName;
     private final boolean isTransient;
     private final ServiceContext serviceContext;
-    private String stateDir;
 
     public QueryCleanupTask(
         final ServiceContext serviceContext,
         final String appId,
+        final Optional<String> topologyName,
         final boolean isTransient,
         final String stateDir
     ) {
       this.serviceContext = Objects.requireNonNull(serviceContext, "serviceContext");
       this.appId = Objects.requireNonNull(appId, "appId");
+      this.topologyName = Objects.requireNonNull(topologyName, "topologyName");
+      queryTopicPrefix = topologyName.map(s -> appId + "-" + s).orElse(appId);
```

Comment:
Would you add a comment here with some context? Just something short about why we concatenate

Reply:
I can add a comment